_posts/2017-5-31-release-1.3.0.md

```diff
@@ -0,0 +1,118 @@
+
+* **Union Operator State**: Flink 1.2.0 introduced broadcast state functionality, but this had not yet been exposed via a public API. Flink 1.3.0 provides the Union Operator State API for exposing broadcast operator state. The union state will send the entire state across all parallel instances to each instance on restore, giving each operator a full view of the state ([FLINK-5991](https://issues.apache.org/jira/browse/FLINK-5991)).
+
+* **Per-Window State**: Previously, the state that a WindowFunction or ProcessWindowFunction could access was scoped to the key of the window but not the window itself. With this new feature, users can keep window state independent of the key ([FLINK-5929](https://issues.apache.org/jira/browse/FLINK-5929)).
+
+# Deployment and Tooling
+
+* **Flink HistoryServer**: Flink’s [HistoryServer](https://ci.apache.org/projects/flink/flink-docs-release-1.3/monitoring/historyserver.html) now allows you to query the status and statistics of completed jobs that have been archived by a JobManager ([FLINK-1579](https://issues.apache.org/jira/browse/FLINK-1579)).
+
+* **Watermark Monitoring in Web Front-end**: For easier diagnosis of watermark issues, the Flink JobManager front-end now provides a new tab to track the watermark of each operator ([FLINK-3427](https://issues.apache.org/jira/browse/FLINK-3427)).
+
+* **Datadog HTTP Metrics Reporter**: Datadog is a widely-used metrics system, and Flink now offers a Datadog reporter that contacts the Datadog http endpoint directly ([FLINK-6013](https://issues.apache.org/jira/browse/FLINK-6013)).
+
+# Table API / SQL
+
+* **Support for Retractions in Table API / SQL**: Retraction is an important building block that will enable a whole range of new applications which require updating previously-emitted results. Examples for such use cases are computation of early results for long-running windows, updates due to late arriving data, or maintaining constantly changing results similar to materialized views in relational database systems. Flink 1.3.0 supports retraction for non-windowed aggregates. Results with updates can be either converted into a DataStream or materialized to external data stores using TableSinks with upsert or retraction support. 
+
+* **Extended support for aggregations in Table API / SQL**: With Flink 1.3.0, the Table API and SQL support many more types of aggregations, including
+	* GroupBy window aggregations (in SQL via the window functions [TUMBLE, HOP, and SESSION windows](https://issues.apache.org/jira/browse/FLINK-6011)) for both batch and streaming.
+
+	* Over window aggregations (only for streaming).
+
+	* Non-windowed aggregations (in streaming with retractions).
+
+	* User-defined aggregation functions for custom aggregation logic.
+
```

Comment:
**External catalog support**: The Table API & SQL allows to register [external catalogs](https://issues.apache.org/jira/browse/FLINK-5568). Table API and SQL queries can then have access to table sources and their schema in the external catalogs without register those tables one by one.